lib/pact_broker/client/tasks/publication_task.rb

```diff
@@ -44,6 +44,14 @@ def branch= branch
         @branch = branch
       end
 
+      def consumer_version=(consumer_version)
+        if consumer_version.nil? && @auto_detect_version_properties
+          @consumer_version = PactBroker::Client::Git.commit(raise_error: true)
+        else
+          @consumer_version = consumer_version
+        end
+      end
+      
```

Comment:
The call out to get the commit from git needs to be done in the getter, not the setter.

```
      def consumer_version
        if @consumer_version.nil? && @auto_detect_version_properties
          @consumer_version = PactBroker::Client::Git.commit(raise_error: true)
        end
      end
```